src/fromager/dependencies.py

```diff
@@ -202,6 +204,44 @@ def default_get_build_sdist_dependencies(
     return hook_caller.get_requires_for_build_wheel()
 
 
+def get_install_dependencies_of_sdist(
+    *,
+    ctx: context.WorkContext,
+    req: Requirement,
+    sdist_root_dir: pathlib.Path,
+    build_env: build_environment.BuildEnvironment,
+) -> set[Requirement]:
+    """Get install requirements (Requires-Dist) from sources
+
+    Uses PEP 517 prepare_metadata_for_build_wheel() API.
+    """
+    pbi = ctx.package_build_info(req)
+    build_dir = pbi.build_dir(sdist_root_dir)
+    logger.info(
+        f"{req.name}: getting install requirements for {req} from sdist in {build_dir}"
+    )
+    pyproject_toml = get_pyproject_contents(build_dir)
+    extra_environ = pbi.get_extra_environ()
+    hook_caller = get_build_backend_hook_caller(
+        build_dir,
+        pyproject_toml,
+        override_environ=extra_environ,
+        network_isolation=ctx.network_isolation,
+        build_env=build_env,
+    )
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        distinfo_name = hook_caller.prepare_metadata_for_build_wheel(tmp_dir)
+        metadata_file = os.path.join(tmp_dir, distinfo_name, "METADATA")
+        with open(metadata_file, "rb") as f:
+            # ignore minor metadata issues
+            metadata = Metadata.from_email(f.read(), validate=False)
```

Comment:
I saw your comment on #561 about why we need to parse the metadata ourselves. We have a few places that do that now. Maybe in a follow-up we can consolidate them into a new helper function?

Reply:
Good idea, I'm going to move loading and parsing bit to a helper function.